R/getAge.R

```diff
@@ -134,84 +130,107 @@ getAge <- function(
-    })
     )
+  }
 
+  # Not sure why entries would have an age but not have final age
+  # 1 California sablefish emailed Brenda on 2023-04-07
+  # 8300 WA records of various spp emailed Theresa on 2021-03-06 and emailed
+  # Kristen H. again on 2023-04-07 to remind her of the discussion where
+  # she had responded and said she was checking with IT
+  all_methods <- Pdata %>%
+    tidyr::unite(
+      all_methods,
+      dplyr::matches("^AGE_METHOD"),
+      sep = "--"
+    ) %>%
+    dplyr::pull(all_methods)
   # Remove Age if none of the available age methods are in keep
-  Age[
-    apply(
-      X = Pdata[, grep("AGE_METHOD[0-9]*$", colnames(Pdata)), drop = FALSE],
-      MARGIN = 1,
-      FUN = function(x) !any(x %in% keep)
-      )
-    ] <- NA
+  out <- dplyr::case_when(
+    grepl(paste0(keep, collapse = "|"), all_methods) ~ Pdata[[column_with_age]],
+    TRUE ~ NA
+  )
 
   if (verbose) {
-    message("\nAges ranged from ", min(Age, na.rm = TRUE), " to ",
-      max(Age, na.rm = TRUE), " (years).")
+    # Make summary objects
+    table_summary <- dplyr::mutate(
+      .data = Pdata,
+      final_age = out,
+      age = rlang::eval_tidy(dplyr::sym(column_with_age))
+    ) %>%
+      dplyr::count(age, final_age) %>%
+      dplyr::arrange(final_age)
+    table_na <- dplyr::filter(
+      .data = table_summary,
+      is.na(final_age) &
+      !is.na(age)
+    ) %>%
+      dplyr::select(-final_age)
+    age_range <- paste(range(out, na.rm = TRUE), collapse = "--")
+    # Print messages to users
+    message("\nAge Distribution -")
+    print(
+      dplyr::filter(table_summary, !is.na(final_age)) %>%
+        dplyr::select(, -age)
+    )
     print(summaryAgeMethod(Pdata, verbose = TRUE))
-    badages <- Pdata[!is.na(Pdata[[col.bestage]]), ]
-    agematch <- sapply(1:NROW(badages), FUN = function(x) {
-      match(badages[x, col.bestage], badages[x, grep("age[0-9]*$", colnames(badages))])
-    })
-    message("\n", col.bestage, " matches value(s) for agers 1, 2, ...;",
-      " NA means no exact match.")
-    print(table(agematch, useNA = "always"))
-    badages <- badages[which(is.na(agematch)), ]
-    badages[["mean"]] <- ceiling(
-      apply(badages[, grep("age[0-9]*$", colnames(badages)), drop = FALSE], MARGIN = 1,
-        mean, na.rm = TRUE)
-      )
-    if (length(which(badages[[col.bestage]] != badages[["mean"]])) > 0) {
-      message("Investigate the following SAMPLES for errors because\n",
-        col.bestage, " doesn't match the mean of all reads:")
-      print(badages[
-        which(badages[[col.bestage]] != badages[["mean"]]),
-        grep("FISH_ID|FINAL|age", colnames(badages))]
-        )
-    }
+    message("\nAges Removed From Final Age -")
+    print(table_na)
+    cli::cli_inform(c(
+      "*" = "{.var Pdata} contains ages that were aged with other methods",
+      "(see above for a summary of number of ages removed by age)",
+      "x" = "{sum(table_na[['n']])} ages were removed from the final ages",
```

Comment:
This was my output from canary. This is to tell me how many potential ages Im missing by excluding certain methods. The red X is a little unsettling as it makes me think its a problem. Could we replace this and the green check with something that doesn't convey good or bad? Maybe use a dash "-"? 

• `Pdata` contains ages that were aged with other methods
(see above for a summary of number of ages removed by age)
✖ 0 ages were removed from the final ages
✔ The following ageing methods in were kept: NA, S, B

Reply:
Thanks @brianlangseth-NOAA for the comment. I changed it to always be a check mark because the task is complete. Then, to help individuals understand the code, {cli} uses named vectors to output a bulleted list where the name of each vector determines the symbol. The available options can be found on the documentation for [`cli::cli_bullets()`](https://cli.r-lib.org/reference/cli_bullets.html).

Here are two examples of the new output from `getAge()`
```
> test <- getAge(bds.pacfin,keep = c("B", "S"), verbose=T)
  `getAge()` summary information -
ℹ The distribution (in numbers) for fish aged 0--32 years is 0, 11, 57, 1023, 6790, 17528, 22234, 19810, 13586, 8457, 5487, 3398, 2366, 1505, 996, 601,  
  347, 185, 156, 74, 61, 38, 16, 12, 14, 5, 5, 5, 1, 3, 1, 0, 2
ℹ Age methods S and B were present
ℹ Age methods 'B', or 'S' were desired
ℹ 100 ages used undesired age methods
✔ Number of ages by age (years) changed to `NA_integer_` is age-3 (n = 1), age-4 (n = 11), age-5 (n = 19), age-6 (n = 28), age-7 (n = 22), age-8 (n = 10), age-9 (n = 7), age-10 (n = 1), age-12 (n = 1)
> test <- getAge(bds.pacfin,keep = c("B"), verbose=T)     
  `getAge()` summary information -
ℹ The distribution (in numbers) for fish aged 0--32 years is 0, 0, 4, 82, 1131, 3428, 4772, 4726, 3691, 2446, 1847, 1336, 1091, 729, 541, 360, 233, 139, 
  115, 64, 49, 32, 15, 11, 12, 3, 4, 5, 0, 2, 1, 0, 2
ℹ Age methods S and B were present
ℹ Age methods 'B' were desired
ℹ 78003 ages used undesired age methods
✔ Number of ages by age (years) changed to `NA_integer_` is age-1 (n = 11), age-2 (n = 53), age-3 (n = 942), age-4 (n = 5670), age-5 (n = 14119), age-6 (n = 17490), age-7 (n = 15106), age-8 (n = 9905), age-9 (n
  = 6018), age-10 (n = 3641), age-11 (n = 2062), age-12 (n = 1276), age-13 (n = 776), age-14 (n = 455), age-15 (n = 241), age-16 (n = 114), age-17 (n =
  46), age-18 (n = 41), age-19 (n = 10), age-20 (n = 12), age-21 (n = 6), age-22 (n = 1), age-23 (n = 1), age-24 (n = 2), age-25 (n = 2), age-26 (n =    
  1), age-28 (n = 1), age-29 (n = 1)
```